src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/TeleporterSystem.java

```diff
@@ -57,8 +62,49 @@
     @ReceiveEvent(components = {SetTeamOnActivateComponent.class})
     public void onActivate(ActivateEvent event, EntityRef entity) {
         EntityRef player = event.getInstigator();
-        String team = setPlayerTeamToTeleporterTeam(player, entity);
-        handlePlayerTeleport(player, team);
+        if (isProperTeamSize(entity, player)) {
+            String team = setPlayerTeamToTeleporterTeam(player, entity);
+            handlePlayerTeleport(player, team);
+        }
+    }
+
+    private boolean isProperTeamSize(EntityRef teleporter, EntityRef player) {
+        setGamePrefab();
+        int oppositeTeamCount = 0;
+        int teleporterTeamCount = 0;
+        int maxTeamSizeDifference = gameEntity.getComponent(LASConfigComponent.class).maxTeamSizeDifference;
+        String teleporterTeam = teleporter.getComponent(LASTeamComponent.class).team;
+        Iterable<EntityRef> characters = entityManager.getEntitiesWith(PlayerCharacterComponent.class,
+                LASTeamComponent.class);
+
+        for (EntityRef character : characters) {
+            String otherPlayerTeam = character.getComponent(LASTeamComponent.class).team;
+            if (teleporterTeam.equals(otherPlayerTeam)) {
+                teleporterTeamCount++;
+            } else if (!otherPlayerTeam.equals(LASUtils.WHITE_TEAM)) {
+                oppositeTeamCount++;
+            }
+        }
+        if (teleporterTeamCount - oppositeTeamCount < maxTeamSizeDifference) {
+            teleporterTeamCount++;
+            gameEntity.getComponent(GameStateComponent.class).redTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? teleporterTeamCount : oppositeTeamCount;
+            gameEntity.getComponent(GameStateComponent.class).blackTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? oppositeTeamCount : teleporterTeamCount;
+            return true;
+        } else {
+            gameEntity.getComponent(GameStateComponent.class).redTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? teleporterTeamCount : oppositeTeamCount;
+            gameEntity.getComponent(GameStateComponent.class).blackTeamCount = teleporterTeam.equals(LASUtils.RED_TEAM) ? oppositeTeamCount : teleporterTeamCount;
```

Comment:
why do you need to change the component in this case?

Reply:
I was wondering if it would be necessary to store the team count, as in the case of disconnecting players the stored information would be incorrect. There would have to be some way to update the information regularly or whenever a player disconnects. This is kind of unnecessary but I added it so that the updates could be made more frequently. Should I remove the team count information altogether?

Reply:
As we're back to counting every time a player tries to join and don't re-use this for that purpose (and also for no other purpose at the moment), I'd say yes, remove it again.
If it comes up in the future again, I'd assume it's pretty easy to reintroduce.